src/systemctl/systemctl-edit.c

```diff
@@ -511,7 +511,7 @@ static int find_paths_to_edit(
                                 if (r < 0)
                                         return r;
 
-                                unit_name = tmp_name;
+                                unit_name = TAKE_PTR(tmp_name);
```

Comment:
Please move the declaration of `tmp_name` to this `if` block, and use `free_and_replace()`, otherwise, `unit_name` is leaked.

Reply:
Oh ok.